R/FilterStateChoices.R

```diff
@@ -448,11 +462,15 @@ ChoicesFilterState <- R6::R6Class( # nolint
         id = id,
         function(input, output, session) {
           output$summary <- renderUI({
-            n_selected <- length(self$get_selected())
-            tagList(
-              tags$span(sprintf("%s levels selected", n_selected)),
-              if (self$get_keep_na()) tags$span("NA") else NULL
-            )
+            if (private$is_disabled()) {
+              tags$span("Disabled")
```

Comment:
Seems that:
- ui_summary is identical across classes and could be moved to the FilterState
- srv_summary could be "abstract" and moved to `FilterState`.
```r
output$summary <- renderUI({
  if (private$is_disabled() {
    tags$span("Disabled")
  } else {
    private$summary()
  }
}}
```


- `XyzFilterState` could have a `private$summary` which returns these summary text elements

Reply:
I think this is outside the scope of this issue. I've opened #218 with these comments. Is that OK?